Fixed inconsistency between Laser and Disabler bolt types

RHSvenson opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
<!-- What do you propose to change with your PR? -->

- Fix inconsistency between Dominator/Hybrid Taser and the Disabler/DisablerSMG by changing them all to act like the Dominator/HybridTaser
- Change Starlight laser weapons to actual lasers, with glass penetration and a beam animation, including the destroyer firing modes
- Remove the heat damage from the Dominator/HybridTaser


## Why we need to add this
<!-- What is the reason for adding these changes? Please post links to Discussions as well as Bug Reports here. Please describe how this will change the game balance. -->
To make weapons of the same type behave uniform.

## Media (Video/Screenshots)
<!--
If your PR contains in-game changes you must provide screenshots/videos of the changes.
-->

https://github.com/user-attachments/assets/069e876f-dd70-423e-ab59-4f176a68434a



## Checks
<!-- check boxes for faster reviewing of your PR -->

- [X] I do not require assistance to complete the PR.
- [X] Before posting/requesting review of a PR, I have verified that the changes work.
- [X] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [X] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
<!--
If you want the players to know about changes made in this PR, specify them using the template outside the comment. Short and informative.

:cl: STARLIGHT TEAM
- add: Added Starlight.
- remove: Removed SS13.
- tweak: Changed SS14.
- fix: Fixed Rinary.
-->
:cl: Trosling
- tweak: Removed heat damage from Dominator/Taser to match the disabler and disabler smg
- tweak: Changed Disabler and Disabler SMG projectiles to behave like the Dominator/Taser
- tweak: Changed Starlight laser weapons to visually match normal laser weapons, and go through windows like them
- tweak: Destroyer bolts are now destroyer beams visually, and go through windows like lasers
- fix: Fixed certain Starlight lasers not going through windows

RHSvenson committed:
Fixed incosistency between Laser and Disabler bolt types

Happyrobot33 commented:
des this change the collision of these beams? wasnt the default for these beams to hit glass instead of go through it?

RHSvenson commented:
> des this change the collision of these beams? wasnt the default for these beams to hit glass instead of go through it?

Sort of. There was a Starlight-unique laser-bolt that was set to collide with glass. It did this because it shared a base-entity with the disabler. I've made a separate one, which mimics the wizden RedLaserBeam.

Only the X-01 and Pulse weapons used this starlight-specific bolt, which made them collide with glass. All other laser weapons used Wizden RedLaserBeam, which did not. This is what has been fixed. All laser weapons now fire the same beam, and the destroyer is also a beam, primarily for aesthetics sake, although I understand if this is overreach in the PR.

Happyrobot33 commented:
This does mean there is slight balance changes, ill need to bring this up with staff but otherwise seems fine so far, just want to avoid balance issues

RHSvenson committed:
Fixed missing changes to RedLaserBeam in battery.yml

RHSvenson committed:
Update combat.yml

Same as previous commit, but correcting reference to LaserBolt in combat.yml for mechs too

RHSvenson commented:
> This does mean there is slight balance changes, ill need to bring this up with staff but otherwise seems fine so far, just want to avoid balance issues

Fair enough, it seemed unintentional that there were two kinds of lasers and disablers, hence the PR. For reference, the damage of the beams is still the same as before, meaning the X-01 does slightly more damage than a normal laser rifle, although it now goes through glass and looks cooler.

RHSvenson committed:
Merge branch 'Starlight' into DisablerTaserDamageFix
